drivers/include/ad7746.h

```diff
@@ -0,0 +1,279 @@
+ * @ingroup    drivers_sensors
+ * @ingroup    drivers_saul
+ * @brief      I2C Capacitance-to-digital converter with temperature sensor
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ * @{
+ *
+ * @file
+ * @brief      AD7746 Capacitance-to-digital converter with temperature
+ *             sensor driver
+ *
+ * @author     Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ */
+
+#ifndef AD7746_H
+#define AD7746_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+/**
+ * @brief  AD7746 default address
+ */
+#ifndef AD7746_I2C_ADDRESS
+#define AD7746_I2C_ADDRESS  (0x48)
+#endif
+
+/**
+ * @brief 0fF capacitance code
+ */
+#define AD7746_ZERO_SCALE_CODE  (0x800000LL)
+
+/**
+ * @brief Interval voltage reference expressed in mV
+ */
+#define AD7746_INTERNAL_VREF    (1170)
+
+/**
+ * @brief Maximum value that can be configured into the DACs.
+ */
+#define AD7746_DAC_MAX          (0x7F)
+
+/**
+ * @brief Channel numbers for reading
+ */
+enum {
+    AD7746_READ_CAP_CH = 0, /**< read capacitive channel */
+    AD7746_READ_VT_CH  = 1 /**< read voltage / temperature channel */
+};
+
+/**
+ * @brief   Named return values
+ */
+enum {
+    AD7746_OK          =  0,       /**< everything was fine */
+    AD7746_NOI2C       = -1,       /**< I2C communication failed */
+    AD7746_NODEV       = -2,       /**< no AD7746 device found on the bus */
+    AD7746_NODATA      = -3        /**< no data available */
+};
+
+/**
+ * @brief Voltage / Temperature channel sample rates
+ */
+typedef enum {
+    AD7746_VT_SR_498 = 0, /**< 49.8 Hz */
+    AD7746_VT_SR_312 = 1, /**< 31.2 Hz */
+    AD7746_VT_SR_161 = 2, /**< 16.1 Hz */
+    AD7746_VT_SR_082 = 3  /**<  8.2 Hz */
+} ad7746_vt_sample_rate_t;
+
+/**
+ * @brief Voltage / Temperature channel modes
+ */
+typedef enum {
+    AD7746_VT_MD_DIS   = -1, /**< channel disabled */
+    AD7746_VT_MD_TEMP  = 0,  /**< internal temperature sensor */
+    AD7746_VT_MD_ETEMP = 1,  /**< external temperature sensor (see datasheet) */
+    AD7746_VT_MD_VDD   = 2,  /**< Vdd voltage monitor */
+    AD7746_VT_MD_VIN   = 3   /**< external voltage input (Vin) */
+} ad7746_vt_mode_t;
+
+/**
+ * @brief Capacitive channel sample rate
+ */
+typedef enum {
+    AD7746_CAP_SR_909 = 0, /**< 90.9 Hz */
+    AD7746_CAP_SR_838 = 1, /**< 83.8 Hz */
+    AD7746_CAP_SR_500 = 2, /**< 50.0 Hz */
+    AD7746_CAP_SR_263 = 3, /**< 26.3 Hz */
+    AD7746_CAP_SR_161 = 4, /**< 16.1 Hz */
+    AD7746_CAP_SR_130 = 5, /**< 13.0 Hz */
+    AD7746_CAP_SR_109 = 6, /**< 10.9 Hz */
+    AD7746_CAP_SR_091 = 7  /**<  9.1 Hz */
+} ad7746_cap_sample_rate_t;
+
+/**
+ * @brief Excitation signal outpus configuration
+ */
+typedef enum {
+    AD7746_EXC_A = 0x06,   /**< enable only exc A output */
+    AD7746_EXC_B = 0x09,   /**< enable only exc B output */
+    AD7746_EXC_AB = 0x0A   /**< enable exc A and B outputs */
+} ad7746_exc_config_t;
```

Comment:
If I see it correctly, this is not used anywhere so I'd remove the typedef and well as it's usage in the device descriptor below